### PR TITLE
docs: More clean-ups (smoe:docs_man_adoc_20240813-2)

### PR DESCRIPTION
**Pull request by smoe**
I took my local version of the https://github.com/LinuxCNC/linuxcnc/pull/3066 and rebased it against master. Added another patch to remove empty blanks at the end of lines. Now looks clean, no exact idea what I did differently.

@petterreinholdtsen , @hansu , Please kindly review this one and then I see what I can do to save what is still important n https://github.com/LinuxCNC/linuxcnc/pull/3066 .

Many thanks,
Steffen

### CODE REVIEW

**Comment by smoe** [comment]
The error is not caused by this PR:
```
Skipping disabled test: /home/runner/work/linuxcnc/linuxcnc/tests/save.1
Running test: /home/runner/work/linuxcnc/linuxcnc/tests/single-step
*** /home/runner/work/linuxcnc/linuxcnc/tests/single-step: XFAIL: test run exited with 1
************** result:
        Machine configuration file is 'test.ini'
        Starting LinuxCNC...
        linuxcncsvr (169910) emcsvr: machine 'unknown'  version '1.1'
        linuxcnc TPMOD=tpmod HOMEMOD=homemod EMCMOT=motmod
        milltask (169923) task: machine 'unknown'  version '1.1'
        emc/task/taskclass.cc 187: can't load tool table.
        Found file(LIB): /home/runner/work/linuxcnc/linuxcnc/lib/hallib/core_sim.hal
        USRMOT: ERROR: command 30 timeout
        emcMotionInit: emcTrajInit failed
        task: main loop took 0.018239 seconds
        task: main loop took 0.019244 seconds
        Shutting down and cleaning up LinuxCNC...
        task: 0 cycles, min=0.000004, max=0.019244, avg=-1.000000, 2 latency excursions (> 10x expected cycle time of 0.001000s)
        LinuxCNC terminated with an error.  You can find more information in the log:
            /home/runner/linuxcnc_debug.txt
        and
            /home/runner/linuxcnc_print.txt
        as well as in the output of the shell command 'dmesg' and in the terminal
************** stderr:
        + linuxcnc -r test.ini
        Note: Using POSIX non-realtime
        emc/tooldata/tooldata_common.cc, 304: Failed to open tool table file 'simpockets.tbl': No such file or directory
        emc/tooldata/tooldata_common.cc, 304: Failed to open tool table file 'simpockets.tbl': No such file or directory
        link (updating variable file): No such file or directory
        timeout in wait_complete_step()
        Note: Using POSIX non-realtime
**************
Running test: /home/runner/work/linuxcnc/linuxcnc/tests/source.0
```

**Comment by petterreinholdtsen** [comment]
[Steffen Möller]
> The error is not caused by this PR:

Yes, there is a race condition in the single-step test.  I restarted the
test and now it succeeded.

I do unfortunately not have time to review these changes for a while.  I
hope someone can have a look.

-- 
Happy hacking
Petter Reinholdtsen

File: docs/src/gui/gscreen.adoc [hunk 1]
```diff
@@ -122,7 +122,7 @@ gscreen-gaxis has a sample custom theme that looks for certain named widgets and
 The Estop and machine-on buttons use different colors then the rest of the buttons so that they stand out.
 This is done in the handler file by giving them specific names an by adding specific commands in the theme's gtkrc file.
 For some info on GTK theming (The sample theme uses the pixmap theme engine), see:
-https://wiki.gnome.org/Attic/GnomeArt/Tutorials/GtkThemes[GTK Themes], 
+https://wiki.gnome.org/Attic/GnomeArt/Tutorials/GtkThemes[GTK Themes],
```
Comment (hansu) [inline docs/src/gui/gscreen.adoc:125]: This space should stay

File: docs/src/code/building-linuxcnc.adoc [hunk 2]
```diff
@@ -131,11 +131,11 @@ Preempt-RT::
...
   LinuxCNC can also be built and run on non-realtime platforms,
   such as a regular install of Debian or Ubuntu without any special realtime kernel.
-.  
+.
   In this mode LinuxCNC is not useful for controlling machine tools,
   but it is useful for simulating the execution of G-code and for testing the non-realtime parts of the system
   (such as the user interfaces, and some kinds of components and device drivers).
-.  
+.
   To make use of the realtime capabilities of LinuxCNC, certain parts of LinuxCNC need to run with root privileges.
```
Comment (hansu) [inline docs/src/code/building-linuxcnc.adoc:139]: The dots doesn't make sense here at all.

File: docs/src/man/man1/linuxcncrsh.1.adoc [hunk 3]
```diff
@@ -48,7 +47,7 @@ All features of LinuxCNC are available via the *linuxcncrsh* interface.
 
 *-ini* _LINUXCNC_INI_FILE_::
   LinuxCNC INI file to use. The -ini option *must* be preceded by two dashes: "--".
-  Defaults to emc.ini if omitted.
+  Defaults to 'emc.ini' if omitted.
```
Comment (hansu) [inline docs/src/man/man1/linuxcncrsh.1.adoc:50]:
```suggestion
  Defaults to `emc.ini` if omitted.
```

**Review by hansu — CHANGES_REQUESTED** [review]
What is CMS_buffer.eps?

Please also remove the asterisks from linuxcncrsh.1.adoc:58 while you are at it :)


```diff
To use linuxcncrsh instead of a normal LinuxCNC GUI like AXIS or Touchy,
specify it in your INI file like this:

----
-*[DISPLAY]*
+[DISPLAY]

-*DISPLAY=linuxcncrsh*
+DISPLAY=linuxcncrsh
----
```

**Comment by smoe** [comment]
> What is CMS_buffer.eps

It is the DIA (https://en.wikipedia.org/wiki/Dia_(software))-generated post script from which https://github.com/LinuxCNC/linuxcnc/blob/master/docs/src/code/CMS_buffer.png is derived. The EPS file itself is not used.

**Comment by smoe** [comment]
Looking at the bad quality of the PNG, it may be worthwhile to have a second look at my idea to remove the EPS.

**Comment by andypugh** [comment]
Would changing the EPS to an SVG work? (Do our tools display SVG? Web browsers display them by default)

**Comment by smoe** [comment]
> Would changing the EPS to an SVG work?
Yes, nice idea, GIMP can do that, I am sure. Let me provide that alternative - and please also tell me if anything else should be changed in that image at that opportunity :-)

> (Do our tools display SVG? Web browsers display them by default)
I do not know, but the PNG I'll make sure to look better.

**Comment by smoe** [comment]
> Would changing the EPS to an SVG work? (Do our tools display SVG? Web browsers display them by default)
The SVG looks nice in inkscape and the PNG is also refreshed.
Have rebased to master, have addressed all of Hansu's changes, except for the one still disputed newline one :-) , and added some more lines on the introduction of linuxcncrsh.1 (feeling that tool to be a bit underrated).

File: docs/src/gui/gscreen.adoc (hunk 1 above)
Comment (hansu) [reply docs/src/gui/gscreen.adoc:125]: This one is still remaining

**Comment by smoe** [comment]
The error in debian:bookworm is triggered by the tests:
```
Running test: ./tests/single-step
*** ./tests/single-step: XFAIL: test run exited with 1
************** result:
        Machine configuration file is 'test.ini'
        Starting LinuxCNC...
        linuxcncsvr (224663) emcsvr: machine 'unknown'  version '1.1'
        linuxcnc TPMOD=tpmod HOMEMOD=homemod EMCMOT=motmod
        milltask (224675) task: machine 'unknown'  version '1.1'
        emc/task/taskclass.cc 187: can't load tool table.
        Found file(LIB): /usr/share/linuxcnc/hallib/core_sim.hal
        USRMOT: ERROR: command 30 timeout
        emcMotionInit: emcTrajInit failed
        task: main loop took 0.018326 seconds
        task: main loop took 0.018875 seconds
        Shutting down and cleaning up LinuxCNC...
        task: 0 cycles, min=0.000004, max=0.018875, avg=-1.000000, 2 latency excursions (> 10x expected cycle time of 0.001000s)
        LinuxCNC terminated with an error.  You can find more information in the log:
            /home/testrunner/linuxcnc_debug.txt
        and
            /home/testrunner/linuxcnc_print.txt
        as well as in the output of the shell command 'dmesg' and in the terminal
************** stderr:
        + linuxcnc -r test.ini
        ERROR: ld.so: object 'libeatmydata.so' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
        Note: Using POSIX non-realtime
        ERROR: ld.so: object 'libeatmydata.so' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
        emc/tooldata/tooldata_common.cc, 304: Failed to open tool table file 'simpockets.tbl': No such file or directory
        emc/tooldata/tooldata_common.cc, 304: Failed to open tool table file 'simpockets.tbl': No such file or directory
        ERROR: ld.so: object 'libeatmydata.so' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
        ERROR: ld.so: object 'libeatmydata.so' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
        ERROR: ld.so: object 'libeatmydata.so' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
        ERROR: ld.so: object 'libeatmydata.so' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
        link (updating variable file): No such file or directory
        line=7; Xpos=5.00; Ypos=0.12; Zpos=0.00; counter=0
```
Tend to think that this PR is not triggering this.

**Comment by hansu** [comment]
Simply look here: https://linuxcnc.org/docs/devel/html/gui/gscreen.html#_overview

This
>For some info on GTK theming (The sample theme uses the pixmap theme engine), see: [GTK Themes](https://wiki.gnome.org/Attic/GnomeArt/Tutorials/GtkThemes), [Pixmap Theme Engine](https://wiki.gnome.org/Attic/GnomeArt/Tutorials/GtkEngines/PixmapEngine).
2.2. Build a GladeVCP Panel

would turn into
>For some info on GTK theming (The sample theme uses the pixmap theme engine), see: [GTK Themes](https://wiki.gnome.org/Attic/GnomeArt/Tutorials/GtkThemes),[Pixmap Theme Engine](https://wiki.gnome.org/Attic/GnomeArt/Tutorials/GtkEngines/PixmapEngine).
2.2. Build a GladeVCP Panel

**Comment by smoe** [comment]
> Simply look here: https://linuxcnc.org/docs/devel/html/gui/gscreen.html#_overview
> 
> This
> 
> > For some info on GTK theming (The sample theme uses the pixmap theme engine), see: [GTK Themes](https://wiki.gnome.org/Attic/GnomeArt/Tutorials/GtkThemes), [Pixmap Theme Engine](https://wiki.gnome.org/Attic/GnomeArt/Tutorials/GtkEngines/PixmapEngine).
> > 2.2. Build a GladeVCP Panel
> 
> would turn into
> 
> > For some info on GTK theming (The sample theme uses the pixmap theme engine), see: [GTK Themes](https://wiki.gnome.org/Attic/GnomeArt/Tutorials/GtkThemes),[Pixmap Theme Engine](https://wiki.gnome.org/Attic/GnomeArt/Tutorials/GtkEngines/PixmapEngine).
> > 2.2. Build a GladeVCP Panel

No, it would not?  I mean, have you seen that happening? Definitely does not happen in the vscode preview. The patch intends to remove the blank that is still followed by a newline, not also the newline. We have lines ending with commas or periods all the time that have proper white space following in the HTML/pdf. I yet fail to see (and would want to understand) why is this now special.

?´+ü´[ <- baby-cat just contributed some extra characters ]

If you do not ultimately mind, please just accept the patch. I'll make a $100 donation to FreeCAD if there is no white space in the PDF after that comma ... and likely spend the rest of the year to find the bug in the autodoc parser.

**Comment by hansu** [comment]
>No, it would not? 

You are right, sorry. I missed the line break and thought just Github editor did the line break :innocent: 
However I still would not trust the VSCode preview as it targets on asciidoctor not asciidoc-py and differs in some border cases.

**Pull request merged**